src/code/z_vimode.c

```diff
@@ -1,17 +1,214 @@
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_vimode/func_800AC030.s")
+void ViMode_LogPrint(OSViMode* viMode) {
+    OSViMode** viMode_ptr = &viMode;
+#define _LOG(line, fmt_str, val)                 \
+    LogUtils_LogThreadId("../z_vimode.c", line); \
+    osSyncPrintf(fmt_str, val)
+    _LOG(87, "osvimodep = %08x\n", (*viMode_ptr));
+    _LOG(88, "osvimodep->comRegs.ctrl = %08x\n", (*viMode_ptr)->comRegs.ctrl);
```

Comment:
considering the source is right there, I'm assuming it's possible to this like so:
`osViModePtr->comRegs.ctrl`

Reply:
I can try to massage this again -- I agree the whole `OSViMode** viMode_ptr = &viMode;` is weird C code, but I needed to add it to get the asm to match up.